cpp/src/TrinoExchangeSource.cpp

```diff
@@ -161,61 +252,99 @@ void TrinoExchangeSource::processDataResponse(
     for (auto& buf : iobufs) {
       totalBytes += buf->capacity();
       if (!singleChain) {
-        buf->trimStart(16);
         singleChain = std::move(buf);
       } else {
         singleChain->prev()->appendChain(std::move(buf));
       }
     }
-    TrinoExchangeSource::updateMemoryUsage(totalBytes);
-
-    page = std::make_unique<exec::SerializedPage>(
-        std::move(singleChain), [pool = pool_](folly::IOBuf& iobuf) {
-          int64_t freedBytes{0};
-          // Free the backed memory from MemoryAllocator on page dtor
-          folly::IOBuf* start = &iobuf;
-          auto curr = start;
-          do {
-            freedBytes += curr->capacity();
-            // FIXME: Jiguang
-            // pool->free(curr->writableData(), curr->capacity());
-            curr = curr->next();
-          } while (curr != start);
-          TrinoExchangeSource::updateMemoryUsage(-freedBytes);
-        });
+    // TODO: Update memory
+    // TrinoExchangeSource::updateMemoryUsage(totalBytes);
+
+    VELOX_CHECK_GE(singleChain->length(), TRINO_RESULT_RESPONSE_HEADER_LEN);
+
+    const int32_t magic = *reinterpret_cast<const int32_t*>(singleChain->data());
+    const int64_t checkSum = *reinterpret_cast<const int64_t*>(singleChain->data() + 4);
+    const int32_t pageCount = *reinterpret_cast<const int32_t*>(singleChain->data() + 12);
+
+    VELOX_CHECK_EQ(magic, TRINO_RESULT_RESPONSE_HEADER_MAGIC);
+    VELOX_CHECK_EQ(sequence_ + pageCount, ackSequence);
+    singleChain->trimStart(TRINO_RESULT_RESPONSE_HEADER_LEN);
+
+    int32_t remaingPage = pageCount;
```

Comment:
Should we use `pageCount` to check whether the response body contains the same number of pages?
Refer to `io.trino.operator.HttpPageBufferClient.PageResponseHandler#handle` and `io.trino.execution.buffer.PagesSerdeUtil.PageReader`

Reply:
If a ```singleChain``` achieved the end of the buffer before all of the pages were constructed, an exception will be thrown.
```cpp
if (!curr) {
    VLOG(google::ERROR) << "Received page body in TrinoExchangeSource is corrupted.";      
}
...
curr = advanceIOBuf(curr, compressedSize + TRINO_SERIALIZED_PAGE_HEADER_SIZE);
```